ksql-functional-tests/src/test/java/io/confluent/ksql/test/EndToEndEngineTestUtil.java

```diff
@@ -495,31 +459,16 @@ private static KsqlEngine getKsqlEngine(final ServiceContext serviceContext) {
 
   static void shouldBuildAndExecuteQuery(final TestCase testCase) {
 
-
-    try (final ServiceContext serviceContext = getServiceContext();
-        final KsqlEngine ksqlEngine = getKsqlEngine(serviceContext)
-    ) {
-      testCase.initializeTopics(
-          serviceContext.getTopicClient(),
-          fakeKafkaService,
-          serviceContext.getSchemaRegistryClient());
-
-      final TopologyTestDriverContainer topologyTestDriverContainer =
-          buildStreamsTopologyTestDriverContainer(
-              testCase,
-              serviceContext,
-              ksqlEngine);
-
-      testCase.verifyTopology();
-      testCase.processInput(topologyTestDriverContainer, serviceContext.getSchemaRegistryClient());
-      testCase.verifyOutput(topologyTestDriverContainer, serviceContext.getSchemaRegistryClient());
-      testCase.verifyMetastore(ksqlEngine.getMetaStore());
+    try (final TestExecutor testExecutor = new TestExecutor()) {
+      testExecutor.buildAndExecuteQuery(testCase);
     } catch (final RuntimeException e) {
       testCase.handleException(e);
     } catch (final AssertionError e) {
       throw new AssertionError("test: " + testCase.getName() + System.lineSeparator()
           + "file: " + testCase.getTestFile()
           + "Failed with error:" + e.getMessage(), e);
+    } catch (final RestClientException | IOException e) {
+      throw new KsqlException("Test failed failed: " + e.getMessage(), e);
```

Comment:
```suggestion
      throw new KsqlException("Test failed: " + e.getMessage(), e);
```
Also I think it would be nice to print out which test failed, not just the message (just like the above error)

Reply:
Added the test name to the message.